lib/utils/snapshots/segmented_gql_data.dart

```diff
@@ -0,0 +1,8 @@
+import 'package:ardrive/services/arweave/graphql/graphql_api.graphql.dart';
+import 'package:ardrive/utils/snapshots/height_range.dart';
+
+abstract class SegmentedGQLData {
+  abstract final HeightRange subRanges;
+  Stream<DriveEntityHistory$Query$TransactionConnection$TransactionEdge$Transaction>
+      getStreamForIndex(int index);
```

Comment:
You can't actually request an index without consuming everything that comes before the sub-range. Consider refactoring it to `getNextStream` while holding a public integer representing the current index
```suggestion
      getNextStream();
  int get currentIndex;
```

Reply:
95dc0f0cea99030b19426fcdc429ce82badf0c97